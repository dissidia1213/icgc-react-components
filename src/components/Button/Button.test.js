//@flow

import { shallow } from "enzyme";
import React from "react";
import Button from "./Button";
//import "../../setupTests";

describe("Button", () => {

	it("Expect to ", ()=> {

		const mockProps = {
			color: "teal"
		};

		const component = shallow(<Button {...mockProps}>OK</Button>);
		expect(component).toMatchSnapshot();
		expect(component.props().color).toEqual("teal");

	});

	it("button click should call Onclick Method", () => {
		
		const clickFn = jest.fn();

		const component = shallow(<Button onClick={clickFn} />);
		component
		  .find('button#my-button-two')
		  .simulate('click');
		expect(clickFn).toHaveBeenCalled();
	  });	

});
