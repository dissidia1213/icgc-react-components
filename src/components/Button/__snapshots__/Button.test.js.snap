// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Expect to  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="teal"
>
    OK
</Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "button",
      "children": "OK",
      "color": "teal",
      "onClick": [Function],
      "primary": true,
    },
    "ref": null,
    "rendered": "OK",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "button",
        "children": "OK",
        "color": "teal",
        "onClick": [Function],
        "primary": true,
      },
      "ref": null,
      "rendered": "OK",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
